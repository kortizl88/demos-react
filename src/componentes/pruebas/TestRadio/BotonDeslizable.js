import React from 'react'

export const BotonDeslizable = ({ leyenda, inactivo = false, estado,  eventoCambioEstado }) => {
    return (
        <>
            <label>{leyenda}</label>
            <input 
                type='checkbox'
                checked={estado}
                disabled={inactivo}
                onChange={(e) => eventoCambioEstado(e.target.checked)}
            />
        </>
    )
}
