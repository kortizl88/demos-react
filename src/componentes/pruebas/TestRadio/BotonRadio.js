import React from 'react'

export const BotonRadio = ({leyenda, inactivo=false, valor, estado, eventoCambioEstado}) => {
    return (
        <>
            <label>{leyenda}</label>
            <input type="radio"
                    checked={ estado === valor }
                    disabled={inactivo}
                    value={valor}
                    onChange={(e) => eventoCambioEstado(e.target.value)}
                    />  
        </>
    )
}
