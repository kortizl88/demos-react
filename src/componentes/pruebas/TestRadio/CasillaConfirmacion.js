import React from 'react'
import './CasillaConfirmacion.scss';

export const CasillaConfirmacion = ({estado, eventoCambio, identificador, texto, inactivo=false }) => {
    
    const estiloEtiqueta = () => {
        if(inactivo){
            return 'etiqueta inactivo'
        }else{
            return 'etiqueta';
        }
    };

    return (
        <div className='casilla-confirmacion'>
            <label className={estiloEtiqueta()} htmlFor={identificador} >{texto}
                <input 
                    className='control' 
                    type="checkbox" 
                    id={identificador}
                    disabled={inactivo}
                    checked={estado}
                    onChange={()=> eventoCambio(!estado)}/>
                <span className="indicador"/>
            </label>
        </div>
    )
}
