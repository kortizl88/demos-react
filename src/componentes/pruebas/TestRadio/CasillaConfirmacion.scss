.casilla-confirmacion{
    .etiqueta{
        position: relative;
        cursor: pointer;
        font-size: 14px;
        line-height: 14px;
        padding-left: 20px;
        &.inactivo{
            color:#CCCDCE;
        }
        input.control {
            position: absolute;
            opacity: 0;
            cursor: pointer;
            height: 0;
            width: 0;
          }
        .indicador{
            position: absolute;
            top: 1px;
            left: 0;
            height: 12px;
            width: 12px;
            margin-right: 8px;
            background-color: #eee;
            border: 1px solid black;
            &:after {
                content: "";
                position: absolute;
                display: none;
              }
        }

        input.control:checked ~ .indicador {
            background-color: #006341;
        }

        input.control:checked ~ .indicador:after {
            display: block;
        }

        .indicador:after {
            left: 3px;
            top: 0px;
            width: 4px;
            height: 8px;
            border: solid white;
            border-width: 0 2px 2px 0;
            transform: rotate(35deg);
          }

        input.control:disabled ~ .indicador {
            background-color: #fff;
            border-color:#CCCDCE;
        }

    }
}