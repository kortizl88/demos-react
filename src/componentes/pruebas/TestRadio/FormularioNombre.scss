.formulario-nombre{
  display: flex;
  .campo{
    width: 311px;
    margin-right: 18px;
    input{
      width: 100%;
      outline: none;
      border: none;
      &::placeholder{
        color: black;
        opacity: 1;
      }
      &:disabled{
        &::placeholder{
          color: #CCCDCE;
        }
      }
    }
    .borde-inferior{
      height: 7px;
      width: 100%;
      border: 2px solid black;
      border-top: none;
    }
    &.inactivo{
      .borde-inferior{
        border-color: #CCCDCE;
      }
    }
    .verificador{
      margin-top: 10px;
    }
  }
}






