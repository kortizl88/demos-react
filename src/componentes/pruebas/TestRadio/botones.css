.control {
    font-family: arial;
    display: block;
    position: relative;
    padding-left: 30px;
    margin-bottom: 0px;
    padding-top: 0px;
    cursor: pointer;
    font-size: 16px;
}
    .control input {
        position: absolute;
        z-index: -1;
        opacity: 0;
    }
.control_indicator {
    position: absolute;
    top: 1px;
    left: 0;
    height: 20px;
    width: 20px;
    background: #ffffff;
    border: 1px solid #266b19;
    border-radius: undefinedpx;
}
.control:hover input ~ .control_indicator,
.control input:focus ~ .control_indicator {
    background: #ffffff;
}

.control input:checked ~ .control_indicator {
    background: #ffffff;
}
.control:hover input:not([disabled]):checked ~ .control_indicator,
.control input:checked:focus ~ .control_indicator {
    background: #ffffff;
}
.control input:disabled ~ .control_indicator {
    background: #e6e6e6;
    opacity: 1;
    pointer-events: none;
}
.control_indicator:after {
    box-sizing: unset;
    content: '';
    position: absolute;
    display: none;
}
.control input:checked ~ .control_indicator:after {
    display: block;
}
.control-radio .control_indicator {
    border-radius: 50%;
}

.control-radio .control_indicator:after {
    left: 2px;
    top: 2px;
    height: 14px;
    width: 14px;
    border-radius: 50%;
    background: #38662f;
    transition: background 250ms;
}
.control-radio input:disabled ~ .control_indicator:after {
    background: #7b7b7b;
}