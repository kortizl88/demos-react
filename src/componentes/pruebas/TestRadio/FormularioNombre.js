import React, {useState} from 'react';
import './FormularioNombre.scss';
import { CasillaConfirmacion } from './CasillaConfirmacion';

export const FormularioNombre = ({estado, eventoCambio}) => {
    const [controlApellidos, setControlApellidos] = useState({paterno:false, materno:false});

    const cambioApellidoPaterno = (valor) =>{
        setControlApellidos({
            ...controlApellidos,
            paterno: valor
        });

        eventoCambio({
            ...estado,
            materno: ''
        });
    };

    const cambioApellidoMaterno = (valor) =>{
        setControlApellidos({
            ...controlApellidos,
            materno: valor
        });

        eventoCambio({
            ...estado,
            paterno: ''
        });
    };

    const claseCampoInactivo = valor => {
        if(valor){
            return 'campo inactivo';
        }else{
            return 'campo';
        }

    }

    return (
        <div className='formulario-nombre'>
            <div id='contenedor-campo-nombre' className='campo'>
                <input
                    id='campo-nombre' 
                    type='text'
                    placeholder='¿cuál es tu nombre?'
                    value={estado.nombre}
                    onChange={ e => eventoCambio({
                        ...estado,
                        nombre: e.target.value
                    })}/>
                <div className='borde-inferior'/>
                <div className='verificador'>
                    Nombre
                </div>
            </div>
            <div id='contenedor-campo-paterno' className={claseCampoInactivo(controlApellidos.materno)}>
                <input 
                    id='campo-apellido-paterno' 
                    type='text'
                    placeholder='¿cuál es tu apellido paterno?'
                    value={estado.paterno}
                    disabled={controlApellidos.materno}
                    onChange={ e => eventoCambio({
                        ...estado,
                        paterno: e.target.value
                    })}/>
                <div className='borde-inferior'/>
                <div className='verificador'>
                    <CasillaConfirmacion 
                        estado={controlApellidos.paterno}
                        eventoCambio={cambioApellidoPaterno}
                        identificador={'selector-app'}
                        texto='Solo apellido paterno'
                        inactivo={controlApellidos.materno} />
                </div>   
            </div>
            <div id='contenedor-campo-materno' className={claseCampoInactivo(controlApellidos.paterno)}>
                <input
                    id='campo-apellido-materno'  
                    type='text'
                    placeholder='¿cuál es tu apellido materno?'
                    value={estado.materno}
                    disabled={controlApellidos.paterno}
                    onChange={ e => eventoCambio({
                        ...estado,
                        materno: e.target.value
                        })}/>
                <div className='borde-inferior'/>
                <div className='verificador'>
                    <CasillaConfirmacion 
                        estado={controlApellidos.materno}
                        eventoCambio={cambioApellidoMaterno}
                        identificador={'selector-apm'}
                        texto='Solo apellido materno'
                        inactivo={controlApellidos.paterno} />
                </div>  
            </div>
        </div>
    )
}
