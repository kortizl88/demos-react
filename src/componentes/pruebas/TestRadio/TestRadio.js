import React, {useState} from 'react'
import { BotonDeslizable } from './BotonDeslizable';
import { BotonRadio } from './BotonRadio';
import './botones.css'

export const TestRadio = () => {

    
    const [isChecked, setIsChecked] = useState(false);
    const [radio, setRadio] = useState();

    return (
        <div>
            <h1>Pruebas de radios con useState</h1>

            <h2>Check esta checado? : { isChecked ? 'true' : 'false' } </h2>

            <h2>Radio seleccionado : { radio } </h2>

            <form>
                <h3>Check</h3>
                <BotonDeslizable 
                    leyenda='mi check personalizado'
                    estado={isChecked}
                    eventoCambioEstado={setIsChecked}
                    />

                <br/>
                <h3>Radios</h3>
                <BotonRadio 
                    leyenda="Manzana"
                    valor='manzana'
                    estado={radio}
                    eventoCambioEstado={setRadio} />
                <br/>
                <BotonRadio 
                    leyenda="Pera"
                    valor='pera'
                    estado={radio}
                    eventoCambioEstado={setRadio} />
                <br/>
                <BotonRadio 
                    leyenda="Platano"
                    valor='platano'
                    estado={radio}
                    eventoCambioEstado={setRadio} />

                    <br/>
                    <br/>
                    <br/>
                    <div class="control-group">
                        <label class="control control-radio">
                            First radio
                            <input type="radio" name="radio" checked="checked" />
                            <div class="control_indicator"></div>
                        </label>
                        <label class="control control-radio">
                            Second radio
                            <input type="radio" name="radio" />
                            <div class="control_indicator"></div>
                        </label>
                        <label class="control control-radio">
                            Disabled
                            <input type="radio" name="radio2" disabled="disabled" />
                            <div class="control_indicator"></div>
                        </label>
                        <label class="control control-radio">
                            Disabled & checked
                            <input type="radio" name="radio2" disabled="disabled" checked="checked" />
                            <div class="control_indicator"></div>
                        </label>
                    </div>

            </form>
        </div>
    )
}
