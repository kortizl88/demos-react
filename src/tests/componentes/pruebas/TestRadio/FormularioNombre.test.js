import React from "react";
import { mount } from "enzyme";
import { FormularioNombre } from "../../../../componentes/pruebas/TestRadio/FormularioNombre";

describe("Pruebas en <FormularioNombre />", () => {
  const propiedades = {
    estado: { nombre: "", paterno: "", materno: "" },
    eventoCambio: jest.fn(),
  };
  const componente = mount(<FormularioNombre {...propiedades} />);

  test("Debe mostrar correctamente el componente", () => {
    expect(componente).toMatchSnapshot();
    expect(componente.find('#campo-nombre').props().value).toBe('');
    expect(componente.find('#campo-apellido-paterno').props().value).toBe('');
    expect(componente.find('#campo-apellido-materno').props().value).toBe('');
    expect(componente.find('#contenedor-campo-paterno').hasClass('campo inactivo')).toBe(false);
    expect(componente.find('#contenedor-campo-paterno').hasClass('campo')).toBe(true);
    expect(componente.find('#contenedor-campo-materno').hasClass('campo inactivo')).toBe(false);
    expect(componente.find('#contenedor-campo-materno').hasClass('campo')).toBe(true);
    expect(componente.find('#selector-app').exists()).toBe(true);
    expect(componente.find('#selector-apm').exists()).toBe(true);
    expect(componente.find('#selector-app').props().checked).toBe(false);
    expect(componente.find('#selector-apm').props().checked).toBe(false);

  });


  test('Debe llamarse el cambio de estado cuando cambie el valor de los input', () => {
    componente.find('#campo-nombre').simulate('change', {target: { value: 'JUAN'}});
    expect(propiedades.eventoCambio).toHaveBeenCalled();
    expect(propiedades.eventoCambio).toHaveBeenCalledWith({
      ...propiedades.estado,
      nombre: 'JUAN'
    });

    componente.find('#campo-apellido-paterno').simulate('change', {target: { value: 'ENE'}});
    expect(propiedades.eventoCambio).toHaveBeenCalled();
    expect(propiedades.eventoCambio).toHaveBeenCalledWith({
      ...propiedades.estado,
      paterno: 'ENE'
    });

    componente.find('#campo-apellido-materno').simulate('change', {target: { value: 'EME'}});
    expect(propiedades.eventoCambio).toHaveBeenCalled();
    expect(propiedades.eventoCambio).toHaveBeenCalledWith({
      ...propiedades.estado,
      materno: 'EME'
    });

  });

  test('Debe inhabilitarse el campo si se selecciona solo apellido paterno', () => {
    componente.find('#selector-app').simulate("change", {});
    expect(componente.find('#contenedor-campo-materno').hasClass('campo inactivo')).toBe(true);
    expect(componente.find('#contenedor-campo-paterno').hasClass('campo')).toBe(true);
    expect(propiedades.eventoCambio).toHaveBeenCalled();
    expect(propiedades.eventoCambio).toHaveBeenCalledWith(
      {
        ...propiedades.estado,
        materno: ''
      }
    );
  })

  test('Debe inhabilitarse el campo si se selecciona solo apellido materno', () => {
    componente.find('#selector-apm').simulate("change", {});
    expect(componente.find('#contenedor-campo-paterno').hasClass('campo inactivo')).toBe(true);
    expect(componente.find('#contenedor-campo-materno').hasClass('campo')).toBe(true);
    expect(propiedades.eventoCambio).toHaveBeenCalled();
    expect(propiedades.eventoCambio).toHaveBeenCalledWith(
      {
        ...propiedades.estado,
        paterno: ''
      }
    );
  })
  
});