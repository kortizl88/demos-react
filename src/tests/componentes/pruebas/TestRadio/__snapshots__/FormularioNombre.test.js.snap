// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FormularioNombre /> Debe mostrar correctamente el componente 1`] = `ReactWrapper {}`;
